src/Compilers/VisualBasic/Test/Semantic/Semantics/RequiredMembersTests.vb

```diff
@@ -0,0 +1,2451 @@
+
+            Dim vbCode = $"
+Module M
+    Sub Main()
+        Dim t {constructor}
+    End Sub
+End Module"
+
+            Dim comp = CreateCompilation(vbCode, {cComp.EmitToImageReference()})
+            comp.AssertTheseDiagnostics(<expected>
+BC37321: Required member 'Public Overrides Property Prop As Integer' must be set in the object initializer or attribute arguments.
+        Dim t <%= constructor %>
+                     ~~~~~~~~~~~~~~
+                                        </expected>)
+        End Sub
+
+        <Theory>
+        <CombinatorialData>
+        Public Sub EnforcedRequiredMembers_Override_AllSet_01(<CombinatorialValues("As New Derived()", " = new Derived()")> constructor As String)
+            Dim cComp = CreateCSharpCompilationWithRequiredMembers(GetDerivedOverrideDefinition(hasSetsRequiredMembers:=False))
+
+            Dim vbCode = $"
+Module M
+    Sub Main()
+        Dim t {constructor} With {{ .Prop = 1 }}
+    End Sub
+End Module"
+
+            Dim comp = CreateCompilation(vbCode, {cComp.EmitToImageReference()})
+            comp.AssertNoDiagnostics()
+        End Sub
+
+        <Theory>
+        <CombinatorialData>
+        Public Sub EnforcedRequiredMembers_Override_AllSet_02(<CombinatorialValues("As New DerivedDerived()", " = new DerivedDerived()")> constructor As String)
+            Dim cComp = CreateCSharpCompilationWithRequiredMembers(GetDerivedOverrideDefinition(hasSetsRequiredMembers:=False))
+
+            Dim vbCode = $"
+Module M
+    Sub Main()
+        Dim t {constructor} With {{ .Prop = 1 }}
+    End Sub
+End Module"
+
+            Dim comp = CreateCompilation(vbCode, {cComp.EmitToImageReference()})
+            comp.AssertNoDiagnostics()
+        End Sub
+
+        <Theory>
+        <CombinatorialData>
+        Public Sub EnforcedRequiredMembers_Override_HasSetsRequiredMembers(<CombinatorialValues("As New Derived()", " = new Derived()")> constructor As String)
+            Dim cComp = CreateCSharpCompilationWithRequiredMembers(GetDerivedOverrideDefinition(hasSetsRequiredMembers:=True))
+
+            Dim vbCode = $"
+Module M
+    Sub Main()
+        Dim t {constructor}
+    End Sub
+End Module"
+
+            Dim comp = CreateCompilation(vbCode, {cComp.EmitToImageReference()})
+            comp.AssertNoDiagnostics()
+        End Sub
+
+        <Fact>
+        Public Sub EnforcedRequiredMembers_StructureNothing_ImplicitCtor()
+            Dim cComp = CreateCSharpCompilationWithRequiredMembers("
+using System.Diagnostics.CodeAnalysis;
+public struct S
+{
+    public required int F;
+}")
+
+            Dim vbCode = $"
+Module M
+    Sub Main()
+        Dim s As S = Nothing
+    End Sub
+End Module"
+
+            Dim comp = CreateCompilation(vbCode, {cComp.EmitToImageReference()})
+            comp.AssertTheseDiagnostics(<expected>
+BC37321: Required member 'Public F As Integer' must be set in the object initializer or attribute arguments.
```

Comment:
It doesn't look like we produce an error for a similar scenario in C# ([https://sharplab.io/#v2:EYLgtghglgdgNAFxFANnAJiA1AHwAIBMABAMICMAsAFADe1RDReZAbEwCxECy0MAFAEp6jOlUbiiAZSIBnIgF4i6AKYAzCAFcUCANzCGAX2pGq1angDMshACcNAYwRTqo8ZaI3lARw1RP6IlgnADE9KgMgA=](https://sharplab.io/#v2:EYLgtghglgdgNAFxFANnAJiA1AHwAIBMABAMICMAsAFADe1RDReZAbEwCxECy0MAFAEp6jOlUbiiAZSIBnIgF4i6AKYAzCAFcUCANzCGAX2pGq1angDMshACcNAYwRTqo8ZaI3lARw1RP6IlgnADE9KgMgA=)):
```
class C1
{
    static void Main()
    {
        S s = default;
    }
}


public struct S
{
    public required int F;
}
```